functorch/_src/eager_transforms.py

```diff
@@ -243,12 +243,15 @@ def vjp(func: Callable, *primals, has_aux=False):
             primals_out = func(*diff_primals)
 
             if has_aux:
+                if not (isinstance(primals_out, tuple) and len(primals_out) == 2):
+                    raise TypeError("Function output should be a tuple: (output, aux) if has_aux is True")
```

Comment:
Should this be a TypeError? (I don't know). If so, we probably want to make sure we are consistent with this...

In a lot of other places where we complain about the input function being malformed we return RuntimeError right now I think

Reply:
Yes, that's true that in other places we raise RuntimeError for type mismatch. I can make here also RuntimeError to make the code consistent. Later we can switch to TypeError if needed such error granularity.

Reply:
Sounds good